src/cmp.rs

```diff
@@ -0,0 +1,267 @@
+        .collect();
+    let result = common
+        .par_iter()
+        .map(|(name, lhs_gid, rhs_gid)| {
+            let mut lhs_closure: Vec<_> = lhs
+                .gsub()?
+                .closure_glyphs([*lhs_gid].into())?
+                .into_iter()
+                .collect();
+            let mut rhs_closure: Vec<_> = rhs
+                .gsub()?
+                .closure_glyphs([*rhs_gid].into())?
+                .into_iter()
+                .collect();
+            if lhs_closure.len() != rhs_closure.len() {
+                return Ok::<String, IconResolutionError>(name.to_string());
+            }
+            lhs_closure.sort();
+            rhs_closure.sort();
+            for (lhs_gid, rhs_gid) in lhs_closure.iter().zip(rhs_closure.iter()) {
+                if !eq(&lhs_outlines, &rhs_outlines, *lhs_gid, *rhs_gid)? {
+                    return Ok(name.to_string());
+                }
+            }
+            Ok(String::from(""))
+        })
+        .collect::<Result<Vec<_>, _>>()?
+        .into_iter()
+        .filter(|f| !f.is_empty());
+
+    Ok(result.collect())
+}
+
+struct Tables<'a> {
+    gvar: Option<Gvar<'a>>,
+    outlines: OutlineGlyphCollection<'a>,
+}
+
+impl<'a> Tables<'a> {
+    fn new(font: &'a FontRef) -> Result<Tables<'a>, ReadError> {
+        Ok(Tables {
+            gvar: font.gvar().ok(),
+            outlines: font.outline_glyphs(),
+        })
+    }
+}
+
+fn eq(
+    lhs: &Tables,
+    rhs: &Tables,
+    lhs_gid: GlyphId,
+    rhs_gid: GlyphId,
+) -> Result<bool, IconResolutionError> {
+    if lhs.gvar.is_some() ^ rhs.gvar.is_some() {
+        return Err(IconResolutionError::Invalid(String::from(
+            "To diff fonts, they both need to have the
+            same type of glyph variation data (either both with gvar or both without).",
+        )));
+    }
+    let l = lhs.outlines.get(lhs_gid).map(|f| draw_outlines(f));
+    let r = rhs.outlines.get(rhs_gid).map(|f| draw_outlines(f));
+    if l != r {
+        return Ok(false);
+    }
+
+    if let (Some(gvar), Some(other_gvar)) = (&lhs.gvar, &rhs.gvar) {
+        let (data, other_data) = (
+            gvar.glyph_variation_data(lhs_gid)?,
+            other_gvar.glyph_variation_data(rhs_gid)?,
+        );
+        let mut d1 = vec![];
+        for t in data.tuples() {
+            for d in t.deltas() {
+                d1.push((d.position, d.x_delta, d.y_delta));
+            }
+        }
+        let mut i = 0;
+        for t in other_data.tuples() {
+            for d in t.deltas() {
+                if (d.position, d.x_delta, d.y_delta) != d1[i] {
+                    return Ok(false);
+                }
+                i += 1;
+            }
+        }
+        return Ok(d1.len() == i);
+        // Compare intermediate_start and intermediate_end when https://github.com/googlefonts/fontations/pull/982 get released.
+    }
+    Ok(true)
+}
+
+fn draw_outlines(lhs: OutlineGlyph) -> BezPath {
```

Comment:
```suggestion
fn draw_outline(lhs: OutlineGlyph) -> BezPath {
```
The plural made me wonder if this was drawing more than one outline.